pkg/assembler/backends/ent/backend/occurrence.go

```diff
@@ -150,10 +150,14 @@ func (b *EntBackend) IngestOccurrence(ctx context.Context,
 				sql.IsNull(occurrence.FieldSourceID),
 			)
 		} else if subject.Source != nil {
-			srcName, err := upsertSource(ctx, tx, *subject.Source)
+			srcNameID, err := upsertSource(ctx, tx, *subject.Source)
 			if err != nil {
 				return nil, err
 			}
+			srcName, err := sourceNameByID(ctx, tx, *srcNameID)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get Sourcename with ID")
+			}
 			occurrenceCreate.SetSource(srcName)
```

Comment:
Please avoid doing further queries but instead prefer to set directly the ID with something like
```suggestion
			occurrenceCreate.SetSourceID(srcNameID)
```

Please check if this applies to other changes in the PR.

Reply:
Done